Starforge/Editor/UI/Menubar.cs

```diff
@@ -0,0 +1,210 @@
+﻿using ImGuiNET;
+using Microsoft.Xna.Framework.Input;
+using Starforge.Core;
+using Starforge.Core.Interop;
+using Starforge.Editor.Render;
+using System;
+using System.IO;
+
+namespace Starforge.Editor.UI {
+    /// <summary>
+    /// The menubar of the window.
+    /// </summary>
+    public static class Menubar {
+        public static float MenubarHeight { get; private set; }
+
+        public static RenderFlags RerenderFlags { get; private set; } = RenderFlags.All;
+        public static bool View_BGDecals = true;
+        public static bool View_BGTiles = true;
+        public static bool View_FGDecals = true;
+        public static bool View_FGTiles = true;
+        public static bool View_OBTiles = true;
+        public static bool View_Entities = true;
+        public static bool View_Triggers = true;
+
+        public static bool DemoWindow = false;
+        public static bool MetricsWindow = false;
+
+        /// <summary>
+        /// Renders the window menubar.
+        /// </summary>
+        /// <param name="hasEditor">Whether or not the map editor is currently loaded.</param>
+        public static void Render() {
+            if (DemoWindow) ImGui.ShowDemoWindow();
+            if (MetricsWindow) ImGui.ShowMetricsWindow();
+
+            if (!ImGui.BeginMainMenuBar()) return;
+            MenubarHeight = ImGui.GetWindowHeight();
+
+            if (ImGui.BeginMenu("File")) {
+                if (ImGui.MenuItem("New")) New();
+                if (ImGui.MenuItem("Open", "CTRL+O")) Open();
+                if (ImGui.MenuItem("Save", "CTRL+S", false, Engine.MapLoaded && MapEditor.Instance.State.Unsaved)) Save();
+                if (ImGui.MenuItem("Save As", "", false, Engine.MapLoaded)) SaveAs();
+
+                ImGui.EndMenu();
+            }
+
+            if (ImGui.BeginMenu("Edit")) {
+                if (ImGui.MenuItem("Undo", "CTRL+Z", false, Engine.MapLoaded && MapEditor.Instance.State.CanUndo())) MapEditor.Instance.State.Undo();
```

Comment:
Undo() already checks if it can undo itself, this check is redundant

Reply:
CanUndo is still checked here so you can see if you have any changes which can be undone. If CanUndo is false, the menu item is then greyed out so you can see that there are no changes you can undo.